### ISSUE TITLE
A few questions about data processing

### ISSUE THREAD

**Issue by berthaSZ** [gmberton/EigenPlaces#12]
Hi @gmberton,

I have read your paper several times, particularly focusing on Section 3. I also checked your code. However, I am still having a hard time understanding how you annotate the image inputs. Ideally, a simplified "class" definition would be very much appreciated. Here are my questions:

1. Could you please provide a simplified version of your class definition?
2. What does "...thus cells within a subset can be later treated as classes for a classification task." mean?
3. Does each cell represent a unique class? In other words, do we have 50 classes if we have 50 cells? If so, what is the point of having subsets?
4. Is the selection of the focal point (through the parameter D) arbitrary? Is each focal point associated with a unique class? In other words, do we have 50 classes if we have 50 focal points?
5. Section 3.2 states, "Finally, within the given cell i, the images facing the focal point c_i are grouped in a single class and used effectively for training." How can one know if an image includes the focal point in the scene? How do you annotate images that do not include the focal point in the scene? I am aware of the assumptions you explained in Section 3.2, and the new class idea for front-facing images. However, there would still be cases where some images in the same cell do not include (or face) the focal point.
6. Section 3.3 states. "In practice, we select images whose orientation is closest
to alpha_j." How does this make sense? Each image j is of course closest to its own alpha_j. What images do you mean here?
7. What are s, m, y_i, and theta_yi in Equation (3)?
8. What is W_latj in Equation (4)?

I look forward to hearing from you.

Best,
Bertha

**Comment by ga1i13o** [comment]
hi
1. a class is a MxM cell, in which we only consider images looking towards a focal point. the focal point is taken along the second principal component of the SVD decomposition of image coordinates
2.  I recommend that you read the CosPlace paper first. a 'subset' is akin to the concept of Group in cosplace; i.e. a set of classes that are not adjacent, taking one cell every N in both longitudinal and latitudinal directions
3. a cell represents a unique class (provided it has at least `min_img_per_class` samples), the point of having subsets is to have within each subset classes that are not adjacent and thus have no visual overlap. again, same concept of cosplace Groups
4. the focal point is taken along the second principal component at distance D from the mean of the image coordinates of that cell. thus the focal point is different for each classs. D is an hyperparameter
5. the focal point is not a 3D point that we need to make sure is visible in each image. it is taken arbitrarily on the second principal component and we select images based on their heading angle so that they look in the direction of the focal point. in practice it might happen that there are some occlusion and the specific point is not actually visible in all images, but in general there will be visual overlap between the images, and the hyperparameter D is selected so that in practice it corresponds to something to the side of the road (or along the road) 
6. the point is that given image coordinates, we need to take into account heading information, i.e. two images with same coordinates might be looking in opposite directions. in the SF-XL dataset each image is a 360 panorama, so this corresponds to cropping the panorama so that its heading angle corresponds to alpha_j
7. s and m are radius and margin hyperparameters of the CosFace loss. \teta_{y_i} is the angle between the prototype of the ground truth class and the image embedding
8.  W_latj is the prototype of class j

let me know if this clarifies your doubts

**Comment by berthaSZ** [comment]
Hi @ga1i13o,

Thank you so much for your reply. It makes things crystal clear for me.

Best,
Bertha
